api/v1alpha1/database_webhook.go

```diff
@@ -42,65 +44,78 @@ func (r *Database) SetupWebhookWithManager(mgr ctrl.Manager) error {
 
-func instanceSpecDefaulterForCreate(r *Database) {
+func instanceSpecDefaulterForCreate(instance *Instance) {
 
-	if len(r.Spec.Instance.DatabaseNames) == 0 {
-		r.Spec.Instance.DatabaseNames = api.DefaultDatabaseNames
+	if instance.DatabaseNames == nil || len(*instance.DatabaseNames) == 0 {
+		instance.DatabaseNames = &api.DefaultDatabaseNames
 	}
 
-	if r.Spec.Instance.TimeZone == "" {
-		r.Spec.Instance.TimeZone = "UTC"
+	if instance.TimeZone == nil || *instance.TimeZone == "" {
+		utc := common.TIMEZONE_UTC
+		instance.TimeZone = &utc
 	}
 
-	if r.Spec.Instance.Profiles.Compute.Id == "" && r.Spec.Instance.Profiles.Compute.Name == "" {
-		r.Spec.Instance.Profiles.Compute.Name = common.PROFILE_DEFAULT_OOB_SMALL_COMPUTE
+	// initialize Profiles block if it's not provided by the user
+
+	if instance.Profiles == nil {
+		databaselog.Info("Initialzing empty Profiles ...")
+		instance.Profiles = &(Profiles{})
+	}
+
+	if instance.Profiles.Compute.Id == "" && instance.Profiles.Compute.Name == "" {
+		instance.Profiles.Compute.Name = common.PROFILE_DEFAULT_OOB_SMALL_COMPUTE
 	}
 
 	// time machine defaulter logic
 
-	tmInfo := r.Spec.Instance.TMInfo
-	if tmInfo.Name == "" {
-		tmInfo.Name = r.Spec.Instance.DatabaseInstanceName + "_TM"
+	// initialize TM block if it's not provided by the user
+	if instance.TMInfo == nil {
+		databaselog.Info("Initialzing empty TMInfo...")
+		instance.TMInfo = &(DBTimeMachineInfo{})
 	}
 
-	if tmInfo.Description == "" {
-		tmInfo.Description = "Time Machine for " + r.Spec.Instance.DatabaseInstanceName
+	if instance.TMInfo.Name == "" {
+		instance.TMInfo.Name = *instance.DatabaseInstanceName + "_TM"
 	}
 
-	if tmInfo.SnapshotsPerDay == 0 {
-		tmInfo.SnapshotsPerDay = 1
+	if instance.TMInfo.Description == "" {
+		instance.TMInfo.Description = "Time Machine for " + *instance.DatabaseInstanceName
 	}
 
-	if tmInfo.SLAName == "" {
-		tmInfo.SLAName = common.SLA_NAME_NONE
+	if instance.TMInfo.SnapshotsPerDay == 0 {
+		instance.TMInfo.SnapshotsPerDay = 1
 	}
 
-	if tmInfo.DailySnapshotTime == "" {
-		tmInfo.DailySnapshotTime = "04:00:00"
+	if instance.TMInfo.SLAName == "" {
+		instance.TMInfo.SLAName = common.SLA_NAME_NONE
 	}
 
-	if tmInfo.LogCatchUpFrequency == 0 {
-		tmInfo.LogCatchUpFrequency = 30
+	if instance.TMInfo.DailySnapshotTime == "" {
+		instance.TMInfo.DailySnapshotTime = "04:00:00"
 	}
 
-	if tmInfo.WeeklySnapshotDay == "" {
-		tmInfo.WeeklySnapshotDay = "FRIDAY"
+	if instance.TMInfo.LogCatchUpFrequency == 0 {
+		instance.TMInfo.LogCatchUpFrequency = 30
 	}
 
-	if tmInfo.MonthlySnapshotDay == 0 {
-		tmInfo.MonthlySnapshotDay = 15
+	if instance.TMInfo.WeeklySnapshotDay == "" {
+		instance.TMInfo.WeeklySnapshotDay = "FRIDAY"
 	}
 
-	if tmInfo.QuarterlySnapshotMonth == "" {
-		tmInfo.QuarterlySnapshotMonth = "Jan"
+	if instance.TMInfo.MonthlySnapshotDay == 0 {
+		instance.TMInfo.MonthlySnapshotDay = 15
+	}
+
+	if instance.TMInfo.QuarterlySnapshotMonth == "" {
+		instance.TMInfo.QuarterlySnapshotMonth = "Jan"
```

Comment:
How do you feel about extracting instance.TMInfo into a variable and using that instead?

Reply:
I think this is better. TMInfo is a inner attribute of instance spec so I want to keep that "context" everywhere it is used.